### PR TITLE
Add FixAll support of UseRecursivePattern refactoring

### PR DESCRIPTION
**Pull request by mavasani**
This was identified as a high value refactoring for FixAll support as it promotes a new language feature (C# recursive patterns)

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @mikadumont

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -77,6 +81,17 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
...
+            => node switch
+            {
+                BinaryExpressionSyntax(LogicalAndExpression) => true,
+                CasePatternSwitchLabelSyntax { WhenClause: { } whenClause } => true,
+                SwitchExpressionArmSyntax { WhenClause: { } whenClause } => true,
+                WhenClauseSyntax { Parent: CasePatternSwitchLabelSyntax } => true,
+                WhenClauseSyntax { Parent: SwitchExpressionArmSyntax } => true,
+                _ => false
+            };
+
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:94]:
```suggestion
        private static bool IsFixableNode(SyntaxNode node)
            => node switch
            {
                BinaryExpressionSyntax(LogicalAndExpression) => true,
                CasePatternSwitchLabelSyntax { WhenClause: not null } => true,
                SwitchExpressionArmSyntax { WhenClause: not null } => true,
                WhenClauseSyntax { Parent: CasePatternSwitchLabelSyntax } => true,
                WhenClauseSyntax { Parent: SwitchExpressionArmSyntax } => true,
                _ => false
            };

```

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -503,5 +518,44 @@ static bool CanConvertToSubpattern(IdentifierNameSyntax name, SemanticModel mode
...
+            // are processed before the outer nodes to refactor.
+            foreach (var originalNode in nodes.Reverse())
+            {
+                // Only process nodes fully within a fixAllSpan
+                if (!fixAllSpans.Any(fixAllSpan => fixAllSpan.Contains(originalNode.Span)))
+                    continue;
+
+                // Get current root, current node to refactor and semantic model.
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var currentNode = root.GetCurrentNodes(originalNode).SingleOrDefault();
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:547]: you're using SingleOrDefault, but not checking the result?  should we check this for null?  or does GetReplacementFunc already do this?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs (hunk 2 above)
Comment (mavasani) [reply src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:547]: Yes, GetReplacementFunc handles null input.

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:94]: I'll fold this suggestion into the next PR and avoid re-running tests on this PR.

**Pull request merged**